pollen_vision/pollen_vision/camera_wrappers/depthai/cam_config.py

```diff
@@ -109,31 +109,39 @@ def get_calib(self) -> dai.CalibrationHandler:
         """Returns a dai.CalibrationHandler object with all the camera's calibration data."""
         return self.calib
 
-    def get_K_left(self) -> npt.NDArray[np.float32]:
-        """Returns the intrinsic matrix of the left camera."""
-        left_socket = get_socket_from_name("left", self.name_to_socket)
-        left_K = np.array(
+    def get_K(self, cam_name: str = "left") -> npt.NDArray[np.float32]:
+        """Returns the intrinsic matrix of the requested camera."""
+        assert cam_name in list(
+            self.name_to_socket.keys()
+        ), f"Camera {cam_name} not found in the config. Available cameras: {list(self.name_to_socket.keys())}"
+        socket = get_socket_from_name(cam_name, self.name_to_socket)
+        K = np.array(
             self.calib.getCameraIntrinsics(
-                left_socket,
+                socket,
                 self.undistort_resolution[0],
                 self.undistort_resolution[1],
             )
         )
 
-        return left_K
+        return K
+
+    def get_D(self, cam_name: str = "left") -> npt.NDArray[np.float32]:
+        assert cam_name in list(
+            self.name_to_socket.keys()
+        ), f"Camera {cam_name} not found in the config. Available cameras: {list(self.name_to_socket.keys())}"
+        socket = get_socket_from_name(cam_name, self.name_to_socket)
+
+        D = np.array(self.calib.getDistortionCoefficients(socket))
+        return D
+
+    # keeping this for now for compatibility with the old code
```

Comment:
add a self._logger.warning("this function is deprecated. Use get_K("left") ") instead of this warning

Reply:
Done